console-subscriber/src/lib.rs

```diff
@@ -106,9 +124,56 @@ enum Event {
         op: WakeOp,
         at: SystemTime,
     },
+    Resource {
+        id: span::Id,
+        metadata: &'static Metadata<'static>,
+        at: SystemTime,
+        concrete_type: String,
+        kind: resource::Kind,
+    },
+    PollOp {
+        metadata: &'static Metadata<'static>,
+        at: SystemTime,
+        resource_id: span::Id,
+        op_name: String,
+        async_op_id: span::Id,
+        task_id: span::Id,
+        readiness: Readiness,
+    },
+    StateUpdate {
+        metadata: &'static Metadata<'static>,
+        at: SystemTime,
+        resource_id: span::Id,
+        update: AttributeUpdate,
+    },
+    AsyncResourceOp {
+        id: span::Id,
+        metadata: &'static Metadata<'static>,
+        at: SystemTime,
+        source: String,
+    },
 }
 
-#[derive(Clone, Copy, Serialize)]
+#[derive(Debug, Clone)]
+enum Readiness {
+    Pending,
+    Ready,
+}
+#[derive(Debug, Clone)]
```

Comment:
style nit: prefer whitespace between type definitions
```suggestion
}

#[derive(Debug, Clone)]
```